Automattic/jetpack
# AI Assistant: Enable backend prompts for 10% of production sites

## Proposed changes:

* Keep backend prompts enabled for beta sites
* Also enable backend prompts for 10% of production sites (all sites where the `blog_id` finishes with `7`, a random choice)

This way we can have 10% of production sites using the new feature, and can scale to 50% and 100% when we are sure it's working as expected.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test on a site with beta extensions enabled and confirm that it's still using the backend prompts:
   * To check if it's using backend prompt, notice the `jetpack-ai-client:ask-question Asking question:` log message
   * Backend prompts will have messages where `role = jetpack-ai`:

<img width="500" alt="Screenshot 2023-09-27 at 15 24 36" src="https://github.com/Automattic/jetpack/assets/6760046/981b7404-0eed-46eb-b5d1-5b0ac88054be">

* Now test on a site with production extensions enabled:
   * if the site ID finishes with a 7, you will see the backend prompts (`role = jetpack-ai`)
   * if the site ID does not finish with a 7, you will see the client prompts, with `role = system` and `role = user` messages:

<img width="350" alt="Screenshot 2023-09-27 at 15 30 46" src="https://github.com/Automattic/jetpack/assets/6760046/8034a5f5-064f-4401-80fd-b73c540b848d">

* To simplify your testing, you can switch the `7` on the code to the final number of your testing site ID, build the plugin again and test it; do it using the correct final number and a different final number, so you can confirm it switches between client and backend prompts
